src/User.php

```diff
@@ -6829,4 +6850,59 @@ final public function isSubstituteOf(int $users_id_delegator, bool $use_date_ran
 
         return (count($result) > 0);
     }
+
+    /**
+     * Validate password based on security rules
+     *
+     * @param string $password password to validate
+     *
+     * @return bool
+     */
+    public function validatePassword(string $password): bool
```

Comment:
Instead of storing passwords validation errors in a `$this->password_validation_errors` property, I think it could be safer to add them in an array passed by reference to the method. Indeed, in GLPI, instances of `CommonDBTM` classes are often reused and value of this property could, in many edge cases, contains stale information.
```suggestion
    public function validatePassword(string $password, array &$errors = []): bool
```

Reply:
I kinda dislike doing it this way but you are right, `CommonDBTM` classes have too many responsibilities which lead to potential missuses.
Done in 81095a293c.